src/requests/cache/set_add_elements.rs

```diff
@@ -0,0 +1,82 @@
+use crate::cache_client::CacheClient;
+use crate::requests::cache::{MomentoRequest, MomentoResponse};
+use crate::simple_cache_client::prep_request;
+use crate::{CollectionTtl, IntoBytes, MomentoResult};
+use momento_protos::cache_client::SetUnionRequest;
+
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # tokio_test::block_on(async {
+/// use std::time::Duration;
+/// use momento::{CredentialProviderBuilder};
+/// use momento::requests::cache::set_add_elements::SetAddElements;
+///
+/// let credential_provider = CredentialProviderBuilder::from_environment_variable("MOMENTO_API_KEY".to_string())
+///     .build()?;
+/// let cache_name = "cache";
+///
+/// let cache_client = momento::CacheClient::new(credential_provider, Duration::from_secs(5))?;
+///
+/// let set_add_elements_response = cache_client.set_add_elements(cache_name.to_string(), "set", vec!["element1", "element2"]).await?;
+/// assert_eq!(set_add_elements_response, SetAddElements::Success {});
+/// # Ok(())
+/// # })
+/// #
+/// }
+/// ```
+pub struct SetAddElementsRequest<S: IntoBytes, E: IntoBytes> {
+    cache_name: String,
+    set_name: S,
+    elements: Vec<E>,
+    collection_ttl: Option<CollectionTtl>,
+}
+
+impl<S: IntoBytes, E: IntoBytes> SetAddElementsRequest<S, E> {
+    pub fn new(cache_name: String, set_name: S, elements: Vec<E>) -> Self {
+        let collection_ttl = CollectionTtl::default();
+        Self {
+            cache_name,
+            set_name,
+            elements,
+            collection_ttl: Some(collection_ttl),
+        }
+    }
+}
+
+impl<S: IntoBytes, E: IntoBytes> MomentoRequest<SetAddElements> for SetAddElementsRequest<S, E> {
+    fn send(
+        self: Self,
+        cache_client: &CacheClient,
+    ) -> impl std::future::Future<Output = MomentoResult<SetAddElements>> + Send {
+        async move {
+            let collection_ttl = self.collection_ttl.unwrap_or_default();
+            let elements = self.elements.into_iter().map(|e| e.into_bytes()).collect();
+            let set_name = self.set_name.into_bytes();
+            let cache_name = &self.cache_name;
+            let request = prep_request(
+                cache_name,
+                SetUnionRequest {
+                    set_name,
+                    elements,
+                    ttl_milliseconds: cache_client.expand_ttl_ms(collection_ttl.ttl())?,
+                    refresh_ttl: collection_ttl.refresh(),
+                },
+            )?;
+
+            let _ = cache_client
+                .data_client
+                .clone()
+                .set_union(request)
+                .await?
+                .into_inner();
+            Ok(SetAddElements::Success {})
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum SetAddElements {
+    Success {},
+}
```

Comment:
hmm what other possibilities for the response are there than success, given that `send` already returns `MomentoResult` which encapsulates everything other than Success? Shouldn't this be a struct like `pub struct SetAddElements;`?

Reply:
I modeled this after what you did for `Get` a while back.

Are you saying we want to have `enum` types for ones that we know can have more than one possible response type, and non-enum types for ones that we don't (currently) think will have more than one response type? it's mostly just another symmetry question in terms of how it will feel to consume the sdk.

Reply:
Result is already an enum type. Some apis are modeled in such a way that a partial success is still a success. For those, a success disposition enum makes sense. For the rest, I see no clear reason to pollute their result types with enums. Do you?

Reply:
i don't have any reasons that I can argue are overwhelmingly compelling, but my reasons would be:

1. symmetry between the different APIs in the Rust SDK
2. future-proofing in case we did add new response types in the future
3. consistency with how we have modeled this in the other language SDKs

if none of those are compelling to you then we can try a few the way that you are describing and decide if it feels too asymmetrical or not.

Reply:
I changed this in https://github.com/momentohq/client-sdk-rust/pull/175/commits/fb50255179e50212ff6ade6de83f0ed55f0da307 . I'm still on the fence about this but willing to try it out for a while if you feel it's the right way to go.